packages/datadog-instrumentations/src/http/client.js

```diff
@@ -0,0 +1,157 @@
+'use strict'
+
+const url = require('url')
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('../helpers/instrument')
+const shimmer = require('../../../datadog-shimmer')
+
+const log = require('../../../dd-trace/src/log')
+
+const startClientCh = channel('apm:http:client:request:start')
+const asyncEndClientCh = channel('apm:http:client:request:async-end')
+const endClientCh = channel('apm:http:client:request:end')
+const errorClientCh = channel('apm:http:client:request:error')
+
+addHook({ name: 'https' }, http => {
+  patch.call(this, http, 'request')
+  patch.call(this, http, 'get')
+
+  return http
+})
+
+addHook({ name: 'http' }, http => {
+  patch.call(this, http, 'request')
+  patch.call(this, http, 'get')
+
+  return http
+})
+
+function patch (http, methodName) {
+  shimmer.wrap(http, methodName, fn => instrumentRequest(fn))
```

Comment:
You should be able to just pass `instrumentRequest` in here directly, rather than wrapping it in this arrow function.